XenModel/XenSearch/GroupingTypes.cs

```diff
@@ -277,7 +277,7 @@ public override String GroupingName
 
         public override String GetGroupName(Object group)
         {
-            if (!(group is T) || i18ns == null)
+            if (!(group is T) || i18ns == null || !i18ns.ContainsKey((T)group))

```

Comment:
It can cause exception being thrown anytime when the i18ns does not contain the group we are looking at. Despite this should not happen, it might, so I think this check is needed to be on the safe side while running in Release build.

For debug build I have added Debug.Assert() as you suggested - this will be a great way to point developers' attention here should they forgot to define the right i18ns entries (as I had managed to do so - answering your first question here)